src/main/java/io/vertx/ext/hawkular/VertxHawkularOptions.java

```diff
@@ -317,4 +322,19 @@ public VertxHawkularOptions setMetricsBridgeEnabled(boolean metricsBridgeEnabled
     this.metricsBridgeEnabled = metricsBridgeEnabled;
     return this;
   }
+  /**
+   * Set metric that will not be registered. Schedulers will check the set {@code disabledMetricsTypes} when
+   * registering metrics suppliers
+   *
+   * @param metricsType the type of metrics
+   * @return the current {@link VertxHawkularOptions} instance
+   */
+  public VertxHawkularOptions disabledMetricsType(MetricsTypeEnum metricsType) {
```

Comment:
Vert.x can be used with different languages (Javascript, Ruby, Groovy for instance). The vert.x stack has a module called vertx-gen which generates idiomatic code. See https://github.com/vert-x3/vertx-codegen
If you write your class as a Java bean (get/set pairs for properties), vertx-gen can find the properties automatically.
Hope it clarifies. Otherwise feel free to ask me by email.